Improvements to command line interface

davidwyld opened the pull request:
- Following @fredblain's request for command line commands à la `dq help` or `dq train help` to work analogously to `dq --help`, `dq -h` or `dq train --help`.

 - Also, have suppressed some numpy warnings in `deepquest/__init__.py` so that they don't appear at the top of running the CLI when typing `dq help` or something.

 - Via this, have discovered that because of `deepquest/__init__.py` which looks like:
```python
from deepquest.train import main as train
from deepquest.predict import main as predict
from deepquest.score import main as score
```
 all of train, predict and score are imported every time deepquest is run. These commands are done in order to tidy up the module naming for running as a python module so we have `dq.train()` instead of `dq.train.main()` for instance.

Any suggestions as to how to clean this up gratefully received. Short of moving all of the functions into one file again..

Commit:
add CLI option for dq help

Commit:
add dq train help option

Commit:
suppress future warnings in numpy for cli

fredblain commented:
>     * Via this, have discovered that because of `deepquest/__init__.py` which looks like:
> 
> 
> ```python
> from deepquest.train import main as train
> from deepquest.predict import main as predict
> from deepquest.score import main as score
> ```
> 
> all of train, predict and score are imported every time deepquest is run. These commands are done in order to tidy up the module naming for running as a python module so we have `dq.train()` instead of `dq.train.main()` for instance.
> 
> Any suggestions as to how to clean this up gratefully received. Short of moving all of the functions into one file again..

I've moved all imports from `__init__.py` into their respective functions in `__main__.py`, and add
```
from .__main__ import *
```
It's quite dirty, but I'm able to call `dq -h` without loading the backend and similarly, to load deepQuest as a Python module, and call train as follows:
```
(dq) ~ ❯❯❯ ipython
Python 3.7.6 (default, Jan  8 2020, 19:59:22) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.12.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import deepquest as dq                                                                                

In [2]: dq.train()                                                                                            
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-0994c7755273> in <module>
----> 1 dq.train()

TypeError: train() missing 1 required positional argument: 'args'
```

davidwyld commented:
I'll have a think about this, at the moment this helps with the command line tool but doesn't serve the intended function for using `dq.train()` from python because of the way the argument handling is done.
By which I mean that `__main__.py`'s job is specifically to act as the command line front-end and move command line arguments into variables used by the core python functions.

Commit:
Move imports in __init__.py to clean up CLI

Move the imports inside functions in __init__.py to prevent importing all of
deepquest, keras and tensorflow when running the command line tool
'dq help' or similar. Essentially just puts off the import of deepquest's
dependencies until one of the functions is run.
Also does the job of providing a clean namespace, ie deepquest.train(),
deepquest.predict() and deepquest.score().

davidwyld commented:
Okay @fredblain , having spoken with @drj11 I've now moved the imports inside functions in `__init__.py` ie we have:
```python
def train(config=None, changes={}):
    import deepquest.train
    train.main(config, changes)
```
see commit  01a14db 

this does the job of tidying up the package's namespace so we have `deepquest.train()` rather than `deepquest.train.main()`
as well as meaning that the whole package and its dependencies don't get imported when running a command line command such as `deepquest help` so warnings and `Using TENSORFLOW!` don't get printed and delay the help text.

davidwyld commented:
@fredblain I'm going to merge this today if you don't have any objections

fredblain reviewed (approved):
Looks good, but I haven't tested.
If it behaves the way we discussed, then it's fine for me.

drj11 reviewed (approved):
Haven't tested it, but it looks good.
